### ISSUE TITLE
Image upload in admin widget?

### ISSUE THREAD

**Issue by MHellmund** [adi-/django-markdownx#28]
Is the file upload magic supposed to work in the admin interface? 
That would be great but I get:

Internal Server Error: /markdownx/upload/
Traceback (most recent call last):
  File "/opt/env/lib/python3.5/site-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/opt/env/lib/python3.5/site-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/env/lib/python3.5/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/opt/env/lib/python3.5/site-packages/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/opt/env/lib/python3.5/site-packages/django/views/generic/edit.py", line 221, in post
    if form.is_valid():
  File "/opt/env/lib/python3.5/site-packages/django/forms/forms.py", line 161, in is_valid
    return self.is_bound and not self.errors
  File "/opt/env/lib/python3.5/site-packages/django/forms/forms.py", line 153, in errors
    self.full_clean()
  File "/opt/env/lib/python3.5/site-packages/django/forms/forms.py", line 363, in full_clean
    self._clean_form()
  File "/opt/env/lib/python3.5/site-packages/django/forms/forms.py", line 390, in _clean_form
    cleaned_data = self.clean()
  File "/opt/env/lib/python3.5/site-packages/markdownx/forms.py", line 45, in clean
    upload = self.cleaned_data['image']
KeyError: 'image'
[09/Aug/2016 17:13:23] "POST /markdownx/upload/ HTTP/1.1" 500 14558

**Comment by adi-** [comment]
Hmmm, strange. What Django version u have?

This could be a problem of not using .get() on cleaned_data. I will try to investigate.

**Comment by MHellmund** [comment]
django 1.9.9

btw, I also get the warning:
/opt/env/lib/python3.5/site-packages/markdownx/widgets.py:34: RemovedInDjango110Warning: render() must be called with a dict, not a Context.
  return t.render(c)

**Comment by adi-** [comment]
I have checked the app with Django 1.10 and python 3.5.2. Everything works as it should be.

**Comment by adi-** [comment]
> /opt/env/lib/python3.5/site-packages/markdownx/widgets.py:34: RemovedInDjango110Warning: render() must be called with a dict, not a Context.
return t.render(c

This one was fixed.

**Comment by adi-** [comment]
I have just spotted an error, when uploaded image is somehow corrupted. I also get this error:

```
upload = self.cleaned_data['image']
KeyError: 'image'
```

Maybe you had the same problem?

**Comment by MHellmund** [comment]
It now seems to me that Linux has a problem with drag&drop to remote X applications. 
I did something like
> ssh -x django-server
> manage.py runserver &
> chromium localhost:8000 &

and got this error when dragging images from the local file manager to the remote chromium window.

But  when using "manage.py runserver 0.0.0.0:8000" and a local chromium, drag&drop works fine. 
So I no longer think that my original problem was due to your package.
